src/studio/src/designer/frontend/shared/components/FileSelector.tsx

```diff
@@ -0,0 +1,133 @@
+import { Description } from '@material-ui/icons';
+import * as React from 'react';
+import { Button, StyledComponentProps, withStyles } from '@material-ui/core';
+import classNames from 'classnames';
+import { getLanguageFromKey } from '../utils/language';
+import theme from '../theme/altinnStudioTheme';
+
+interface IFileSelectorState {
+  selectedFileName: string | undefined;
+}
+interface IFileSelectorProps extends StyledComponentProps {
+  language: any,
+  submitHandler: (file: FormData, fileName: string) => void;
+  busy: boolean;
+  labelTextResource: string;
+  formFileName: string;
+  accept?: string;
+}
+const styles = {
+  root: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    '&>*': { margin: 4, height: 36 },
+    '& input:focus-visible + label': {
+      outline: '2px solid black',
+    },
+    '& label': {
+      flex: 1,
+      display: 'inline-flex',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      '& span': {
+        flex: 1,
+        height: 36,
+        padding: 2,
+        paddingTop: 5,
+        boxSizing: 'border-box',
+        border: '1px solid #0062BA',
+        '&.empty': {
+          fontStyle: 'italic',
+          color: 'grey',
+        },
+      },
+    },
+    '& button': {
+      borderBottom: `1px solid ${theme.altinnPalette.primary.blueDark}`,
+      color: theme.altinnPalette.primary.white,
+      background: theme.altinnPalette.primary.blueDark,
+      textTransform: 'none' as 'none',
+      fontSize: 16,
+      fontWeight: 400,
+      borderRadius: '0',
+      '&:hover': {
+        background: theme.altinnPalette.primary.blueDarker,
+        color: theme.altinnPalette.primary.white,
+      },
+      '&:focus': {
+        background: theme.altinnPalette.primary.blueDarker,
+        color: theme.altinnPalette.primary.white,
+      },
+      '&:disabled': {
+        borderBottom: `1px solid ${theme.altinnPalette.primary.greyMedium}`,
+      },
+    },
+  },
+};
+class FileSelector extends React.Component<IFileSelectorProps, IFileSelectorState> {
```

Comment:
Class component? Is this an intentional choice?

Reply:
Yes, it is to keep the file input as a ref.
`React.createRef()` can only be used in a class component.
I'd rather have this written as a functional component, but couldn't find a way of doing it.

Reply:
Scratch that. I figured out that I didn't need to use `React.createRef()`

Reply:
Nice 👍